### PR TITLE
Make the NumEpisodesObserver count all episodes in a batch of trajectories  

### PR DESCRIPTION
**Pull request by leroidauphin**
The `NumStepsTransitionObserver` counts the number of steps in a batch of trajectories by using a `reduce_sum` across the batch dimension. By contrast, the `NumEpisodesObserver` only counts the number of episodes in the zero-th indexed member of the batch. 

This PR modifies the `NumEpisodesObserver` to follow the pattern established in the `NumStepsTransitionObserver` and use `reduce_sum` to count episodes across all of the episodes in the batch of trajectories.

### CODE REVIEW

**Comment by leroidauphin** [comment]
@googlebot I signed it!

**Comment by egonina** [comment]
Thanks for the PR! Would you mind adding a unit test to ensure this behaves as intended?

**Comment by leroidauphin** [comment]
> Thanks for the PR! Would you mind adding a unit test to ensure this behaves as intended?

My apologies - I have now added a unit test.

File: tf_agents/drivers/test_utils_test.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      action=np.random.rand(batch_size, traj_len),
+      policy_info=(),
+      reward=np.random.rand(batch_size, traj_len),
+      discount=np.ones((batch_size, traj_len)),
+      step_type=np.zeros((batch_size, traj_len)),
+      next_step_type=next_step_type)
+
+    observer = driver_test_utils.NumEpisodesObserver()
+    observer(traj)
+    assert observer.num_episodes.read_value() == batch_size
```
Comment (egonina) [inline tf_agents/drivers/test_utils_test.py]: you can use self.assertEqual here (https://www.tensorflow.org/api_docs/python/tf/test/TestCase#assertEquals).

**Review by egonina — COMMENTED** [review]
Thanks for adding the test!

File: tf_agents/drivers/test_utils_test.py (hunk 1 above)
Comment (leroidauphin) [reply tf_agents/drivers/test_utils_test.py]: Thanks for the review!

**Pull request merged**